client/components/checkbox/index.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+* External dependencies
+*/
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+/**
+* Interal dependencies
+*/
+import Gridicon from 'components/gridicon';
+
+export default React.createClass( {
+	displayName: 'Checkbox',
+
+	propTypes: {
+		checked: PropTypes.bool.isRequired,
+		disabled: PropTypes.bool,
+		onChange: PropTypes.func
+	},
+
+	getInitialState() {
+		return {
+			focused: false
+		}
+	},
+
+	getDefaultProps() {
+		return {
+			disabled: false
+		};
+	},
+
+	handleChange( e ) {
+		const { onChange } = this.props;
+
+		onChange && onChange( e, this.state.checked );
```

Comment:
This is a case where we have a bunch of code just to make sure we don't call a function that doesn't exist. If, however, we added a default `onChange` handler, we could know that we could always call this.

```js
import noop from 'lodash/utility/noop';

getInitialProps() {
	return {
		disabled: false,
		onChange: noop
	};
}

handleChange( e ) {
	this.props.onChange( e, this.state.checked );
}
```